h/views/api/config.py

```diff
@@ -61,30 +62,18 @@ def add_api_view(
                                   `route_name` must be specified.
     :param dict settings: Arguments to pass on to ``config.add_view``
     """
-
-    # Get the HTTP method for use in the API links metadata
-    primary_method = settings.get("request_method", "GET")
-    if isinstance(primary_method, tuple):
-        # If the view matches multiple methods, assume the first one is
-        # preferred
-        primary_method = primary_method[0]
-
     settings.setdefault("accept", "application/json")
     settings.setdefault("renderer", "json")
     settings.setdefault("decorator", cors_policy)
 
     if link_name:
         link = {
```

Comment:
I would consider creating a class for link objects in future so that the parameters of the helper can be documented better than just saying the input `link` argument is a `dict`.

Reply:
Yes, was talking about this exact thing to @seanh yesterday. They should be a class. I wonder if it's worth taking the time to do that now. Maybe a follow-on PR? What do you think?